client/state/themes/selectors.js

```diff
@@ -44,30 +44,28 @@ export const getTheme = createSelector(
 			return null;
 		}
 
-		const theme = manager.getItem( themeId );
-		if ( siteId === 'wpcom' || siteId === 'wporg' ) {
-			return theme;
-		}
-
-		if ( ! theme ) {
-			return null;
-		}
-
-		// We're dealing with a Jetpack site. If we have theme info obtained from the
-		// WordPress.org API, merge it.
-		const wporgTheme = getTheme( state, 'wporg', themeId );
-		if ( ! wporgTheme ) {
-			return theme;
-		}
-		return {
-			...theme,
-			...pick( wporgTheme, [ 'demo_uri', 'download', 'taxonomies' ] )
-		};
+		return manager.getItem( themeId );
 	},
 	( state ) => state.themes.queries
 );
 
 /**
+ * Returns a theme object from what is considered the 'canonical' source, i.e.
+ * the one with richest information. Checks WP.com (which has a long description
+ * and multiple screenshots, and a preview URL) first, then WP.org (which has a
+ * preview URL), then the given JP site.
+ *
+ * @param  {Object}  state   Global state tree
+ * @param  {Number}  siteId  Jetpack Site ID to fall back to
+ * @param  {String}  themeId Theme ID
+ * @return {?Object}         Theme object
+ */
+export function getCanonicalTheme( state, siteId, themeId ) {
+	const source = find( [ 'wpcom', 'wporg', siteId ], s => getTheme( state, s, themeId ) );
```

Comment:
Sure. `getTheme` will work with undefined without a problem so `find` is OK. And my suggestion saves one call to `getTheme` but that may be in fact unnecessary optimization.